src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -396,6 +400,14 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
                 ? Telemetry.SelectLocalJupyterKernel
                 : Telemetry.SelectRemoteJupyterKernel;
             sendKernelTelemetryEvent(document.uri, telemetryEvent);
+            window.visibleNotebookEditors
+                .filter((editor) => editor.document === document)
+                .forEach((editor) =>
+                    controller.postMessage(
+                        { message: IPyWidgetMessages.IPyWidgets_onKernelChanged, payload: undefined },
```

Comment:
Is it ok that at this point we've not actually started / changed the kernel under the covers? This is just the UI selection until they actually run.

Reply:
Yes, this is fine.